core/isolate.rs

```diff
@@ -315,19 +317,25 @@ impl Isolate {
       PinnedBuf::new(zero_copy_buf),
     );
 
-    let op = match op {
-      Op::Async(mut fut) => {
-        // Tries to greedily poll async ops once. Often they are immediately ready, in
-        // which case they can be turned into a sync op before we return to V8. This
-        // can save a boundary crossing.
-        #[allow(clippy::match_wild_err_arm)]
-        match fut.poll() {
-          Err(_) => panic!("unexpected op error"),
-          Ok(Ready(buf)) => Op::Sync(buf),
-          Ok(NotReady) => Op::Async(fut),
+    let op = if isolate.eager_count != 50 {
+      isolate.eager_count += 1;
+      match op {
+        Op::Async(mut fut) => {
+          // Tries to greedily poll async ops once. Often they are immediately ready, in
+          // which case they can be turned into a sync op before we return to V8. This
+          // can save a boundary crossing.
+          #[allow(clippy::match_wild_err_arm)]
+          match fut.poll() {
+            Err(_) => panic!("unexpected op error"),
+            Ok(Ready(buf)) => Op::Sync(buf),
+            Ok(NotReady) => Op::Async(fut),
+          }
         }
+        Op::Sync(buf) => Op::Sync(buf),
       }
-      Op::Sync(buf) => Op::Sync(buf),
+    } else {
+      isolate.eager_count = 0;
```

Comment:
I think resetting this to 0 should be done when we poll the pending_ops stream. Otherwise the 50th op will not be polled eagerly will but the 51st will.

Reply:
Thanks, you're right! It's even better latency now :tada:

```
$ third_party/wrk/mac/wrk -d 10s --latency http://127.0.0.1:4544
Running 10s test @ http://127.0.0.1:4544
  2 threads and 10 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    97.23us   40.92us   2.09ms   92.81%
    Req/Sec    50.06k     3.38k   52.60k    89.11%
  Latency Distribution
     50%   87.00us
     75%  101.00us
     90%  109.00us
     99%  258.00us
  1005713 requests in 10.10s, 48.92MB read
Requests/sec:  99578.13
Transfer/sec:      4.84MB
```